### DISCUSSION TITLE
EMQX QUIC cert_required

### DISCUSSION THREAD

**Discussion by esefberfu** [emqx/emqx#10973] [Q&A]
Hello guys,
I enables quic on emqx server and [create self-signed certificate](https://www.emqx.com/en/blog/enable-two-way-ssl-for-emqx)
I create QUIC listener and I put emqx.pem to TLS CRT, emqx.key to TLS Key, ca.pem to CA Cert in quic listener config using UI Panel.

I am running below command on emqtt_bench for testing.
`
 ./emqtt_bench pub --quic true --host broker_host_name -p 14567 -t topic1/test -u testuser -P 123456-m '{"data":123}' -I 1 -i 10 -c 1 --certfile ./client.pem --keyfile ./client.key
`

but connection fail and return **cert_required** error message.

ca.pem -> subject=C = CN, ST = Zhejiang, L = Hangzhou, O = EMQX, CN = CA
emqx.pem -> subject=C = CN, ST = Zhejiang, L = Hangzhou, O = EMQX, CN = CA
client.pem -> subject=C = CN, ST = Zhejiang, L = Hangzhou, O = EMQX, CN = client

**Comment by zmstone** [comment]
Hi @esefberfu 

Could you provide the exact version of `emqtt_bench`.

**Reply by esefberfu** [reply]
I am using this version. (https://github.com/emqx/emqtt-bench/tree/0.4.11)
I added --ssl true parameter to command. Now return bad_certificate error. But before all of that, How to send CA certificate ?. CA certificate is missin in command. Alternatively, I try using MQTTX Client while mqtts protocol. Works fine!

I take same error too on dotnet MQTTnet client library.

**Reply by zmstone** [reply]
Hi @esefberfu 
Just for my understanding: you have tested MQTTX against EMQX's SSL listener which is working fine. However, you cannot make `emqtt-bench` to work with EMQX's QUIC listener.

As for CA certificates. `emqtt-bench` does not support server certificate validation (because almost no one actually need to over-stress the client during server benchmark tests), so there is no need to configure CA certificates.
If you want to test client certificate validation (on the server side), you can provide the client certificate (followed by its intermediate CA certificates if any) in `client.pem`, and put the client's root CA in the server's CA bundle file, (ca.pem).

**Comment by zmstone** [comment]
As for the `cert_required` error, I am not sure what happened in your test environment.
Did you build emqtt-bench from source code? or did you download a pre-built binary package?

**Comment by zmstone** [comment]
Hi again.
`--ssl` option is missing in your `emqtt-bench` command line, could you please add it and test again.
I still cannot reproduce the `cert_required` error though.
